### PR TITLE
DOCS-1290-CE-update-arch-diagram

### PR DESCRIPTION
**Pull request by bmckercher123**
## Add new API for MT

- CE 3.17
Note: No updates to CC arch diagram at this time. Changes are all on the management cluster side. 

Issue:

Link to docs preview:
https://deploy-preview-423--tigera.netlify.app/calico-enterprise/next/reference/architecture/overview#api-gateway

SME review:
[ x ] A SME has approved this change.

DOCS review:

[ x ]  A member of the docs team has approved this change.

### CODE REVIEW

File: calico-enterprise/reference/architecture/overview.mdx [hunk 1]
```diff
@@ -102,9 +102,9 @@ Provides the API for listing, downloading, and rendering reports, and RBAC by pe
...
 
-### Elasticsearch gateway
+### Datastore API
 
-**Main task**: Protects managed clusters from potential attacks in a multi-cluster management (mcm) deployment. Enforces that 1) managed cluster components do not get credentials for an Elasticsearch user (for access/permissions to Elasticsearch and Kibana), and 2) all Elasticsearch requests from a managed cluster are only on an index that the cluster is allowed to access. In short, a compromised component in a managed cluster cannot write data or create indexes for another cluster/tenant.
+**Main task**: A {{prodname}} HTTP API that enforces tenancy boundaries and  prerforms RBAC checks for added security. Protects managed clusters from potential attacks in a multi-cluster management (mcm) deployment. Enforces that 1) managed cluster components do not get credentials for an Elasticsearch user (for access/permissions to Elasticsearch and Kibana), and 2) all Elasticsearch requests from a managed cluster are only on an index that the cluster is allowed to access. In short, a compromised component in a managed cluster cannot write data or create indexes for another cluster/tenant.
```
Comment (asincu) [inline calico-enterprise/reference/architecture/overview.mdx]: Typo: performs

Comment (asincu) [inline calico-enterprise/reference/architecture/overview.mdx]: I would rephrase this to an API gateway instead of an HTTP API. If at some point we want to switch to an alternative protocol, I would not want to change this doc.

File: calico-enterprise/reference/architecture/overview.mdx [hunk 2]
```diff
@@ -25,7 +25,7 @@ Calico open-source components are the foundation of {{prodname}}. {{prodname}} p
 
 - [calicoq](#calicoq)
 - [Compliance](#compliance)
-- [Elasticsearch gateway](#elasticsearch-gateway)
+- [Linseed API](#linseed-api)
```
Comment (asincu) [inline calico-enterprise/reference/architecture/overview.mdx]: Should this be `[Datastore API](#datastore-api)`  ?

File: calico-enterprise/reference/architecture/overview.mdx (hunk 1 above)
Comment (asincu) [reply calico-enterprise/reference/architecture/overview.mdx]: I would suggest removing any reference to creating new indices from `In short, a compromised component in a managed cluster cannot write data or create indexes for another cluster/tenant.`

Comment (asincu) [inline calico-enterprise/reference/architecture/overview.mdx]: There is a lot of emphasis on MCM, but we are also protecting any access to Elastic as a data store.

Comment (bmckercher123) [reply calico-enterprise/reference/architecture/overview.mdx]: Fixed.

Comment (bmckercher123) [reply calico-enterprise/reference/architecture/overview.mdx]: Deleted "or create indexes"

Comment (bmckercher123) [reply calico-enterprise/reference/architecture/overview.mdx]: Rewrote to include comment.

File: calico-enterprise/reference/architecture/overview.mdx (hunk 2 above)
Comment (bmckercher123) [reply calico-enterprise/reference/architecture/overview.mdx]: Agreed upon Linseed API/ES gateway for this version.

**Review by asincu — APPROVED** [review]

**Comment by ctauchen** [comment]
LGTM
